drivers/ostree/driver.go

```diff
@@ -0,0 +1,178 @@
+		}
+
+		if err := exec.Command("ostree", fmt.Sprintf("--repo=%s", repoLocation), "init").Run(); err != nil {
+			return nil, fmt.Errorf("Could not create OSTree repository: %v", err)
+		}
+	}
+
+	repo, err := otbuiltin.OpenRepo(repoLocation)
+	if err != nil {
+		return nil, fmt.Errorf("Could not open the OSTree repository: %v", err)
+	}
+
+	d := &Driver{
+		home:    home,
+		overlay: overlay,
+		convert: make(map[string]bool),
+	}
+	d.repo = repo
+	d.repoLocation = repoLocation
+	return graphdriver.NewNaiveDiffDriver(d, uidMaps, gidMaps), nil
+}
+
+// Driver must be wrapped in NaiveDiffDriver to be used as a graphdriver.Driver
+type Driver struct {
+	home         string
+	overlay      graphdriver.Driver
+	repo         *otbuiltin.Repo
+	repoLocation string
+	convert      map[string]bool
+}
+
+func (d *Driver) String() string {
+	return "ostree"
+}
+
+// Status is used for implementing the graphdriver.ProtoDriver interface. OSTREE does not currently have any status information.
+func (d *Driver) Status() [][2]string {
+	return d.overlay.Status()
+}
+
+// Metadata is used for implementing the graphdriver.ProtoDriver interface. OSTREE does not currently have any meta data.
+func (d *Driver) Metadata(id string) (map[string]string, error) {
+	return d.overlay.Metadata(id)
+}
+
+// Cleanup is used to implement graphdriver.ProtoDriver. There is no cleanup required for this driver.
+func (d *Driver) Cleanup() error {
+	return d.overlay.Cleanup()
+}
+
+// CreateReadWrite creates a layer that is writable for use as a container
+// file system.
+func (d *Driver) CreateReadWrite(id, parent string, opts *graphdriver.CreateOpts) error {
+	return d.overlay.CreateReadWrite(id, parent, opts)
+}
+
+// Create prepares the filesystem for the OSTREE driver and copies the directory for the given id under the parent.
+func (d *Driver) Create(id, parent string, opts *graphdriver.CreateOpts) error {
+	d.convert[id] = true
+	return d.overlay.Create(id, parent, opts)
+}
+
+func getDir(home, id string) string {
+	return fmt.Sprintf("%s/%s/diff", home, id)
+}
+
+// Create prepares the filesystem for the OSTREE driver and copies the directory for the given id under the parent.
+func (d *Driver) convertToOSTree(root, id string) error {
+	if _, err := d.repo.PrepareTransaction(); err != nil {
+		return fmt.Errorf("Could not prepare the OSTree transaction: %v", err)
+	}
+
+	commitOpts := otbuiltin.NewCommitOptions()
+	commitOpts.Timestamp = time.Now()
+	commitOpts.Parent = "0000000000000000000000000000000000000000000000000000000000000000"
+	branch := fmt.Sprintf("ocilayer/%s", id)
+
+	if _, err := d.repo.Commit(root, branch, commitOpts); err != nil {
+		return fmt.Errorf("Could not commit the layer: %v", err)
+	}
+
+	if _, err := d.repo.CommitTransaction(); err != nil {
+		return fmt.Errorf("Could not complete the OSTree transaction: %v", err)
+	}
+
+	if err := system.EnsureRemoveAll(root); err != nil {
+		return err
```

Comment:
errors.Wrapf(err, "failed to remove %q", root)

Reply:
I looked at the code and it already gives a similar error message, so I think it would be duplicated `return errors.Wrapf(e, "error while removing %s", dir)`.  In some other cases it returns directly the error from `os.RemoveAll()`